projects/ngx-formentry/src/form-entry/value-adapters/diagnosis.adapter.ts

```diff
@@ -0,0 +1,175 @@
+import {Injectable} from '@angular/core';
+import {Form} from '../form-factory/form';
+import {ValueAdapter} from './value.adapter';
+
+@Injectable()
+export class DiagnosisValueAdapter implements ValueAdapter {
+  formDiagnosisNodes = [];
+
+  generateFormPayload(form: Form) {
+    this.formDiagnosisNodes = [];
+    this._findDiagnosisQuestionNodes(form.rootNode);
+    return this._createDiagnosesPayload(this.formDiagnosisNodes, form.existingDiagnoses);
+  }
+
+  populateForm(form: Form, diagnoses) {
+    form.existingDiagnoses = diagnoses?.filter(d => {
+      return !d.voided;
+    });
+    this.formDiagnosisNodes = [];
+    this._findDiagnosisQuestionNodes(form.rootNode);
+    this._setDiagnosesValues(this.formDiagnosisNodes, form.existingDiagnoses, 1);
```

Comment:
While the 2.x RefApp used primary and secondary diagnoses, the field technically supports any non-zero value for rank... It would be good to be able to handle that...